### DISCUSSION TITLE
`schedule` and `prConcurrentLimit`

### DISCUSSION THREAD

**Discussion by northern-mindset** [renovatebot/renovate#10006] [Request Help] [answered]
Suppose the following scenario:

- Renovate is configured to run monthly.
- Renovate has to create more PRs during the scheduled run than `prConcurrentLimit`.
- It creates only part of them and stops.
- Very soon the PRs will be tested and merged (manually or automerged).
- Therefore, spare “slots” are there to create new PRs.

**When Renovate will create the rest of the pending PRs? The next month or soon after the previous PRs get merged/closed?** The same question goes for `prHourlyLimit`.

**Comment by rarkins** [comment, marked as answer]
The next month, assuming that the scheduled window has now passed by the time the PRs are merged/closed.

**Comment by northern-mindset** [comment]
Can you please elaborate on what is the “scheduled window” in this context?

In our case Ci cycle takes a couple of minutes, and automerge is configured
for the repository. How can I know whether this will fall into the
scheduled window?

**Reply by rarkins** [reply]
In referring to the "schedule" field which you can read the documentation for. Renovate needs at least two runs to automerge - first one creates the PR and second one merges it if all checks have passed. If you are running monthly and hoping to automerge then I recommend a wide schedule (eg all day on first day of the month).

**Reply by northern-mindset** [reply]
Oh, I see. I didn't realize that the schedule defines a time range, not a particular point in time. I'll give it a try. Perhaps the `on the first day of the month` schedule should work, as it gives the whole 24 hours to complete the cycle of creating new PRs. The only thing is that any failed PR can block creating others. But still this is solvable with the dashboard.
